### PR TITLE
added `center`, `det`, `is_rational`, `kernel`, `order` for group characters

### PR DESCRIPTION
**Pull request by ThomasBreuer**
This addresses issue #1875 and some comments about pull request #1926.

### CODE REVIEW

File: src/Groups/group_characters.jl [hunk 1]
```diff
@@ -1207,7 +1229,7 @@ Base.iterate(chi::GAPGroupClassFunction, state = 1) = state > length(chi.values)
 
 @doc Markdown.doc"""
     degree(::Type{T} = QQFieldElem, chi::GAPGroupClassFunction)
-           where T <: Union{IntegerUnion, ZZRingElem, mpq, QQAbElem}
+           where T <: Union{IntegerUnion, ZZRingElem, QQFieldElem, QQAbElem}
```
Comment (thofma) [inline src/Groups/group_characters.jl:1214]: Not related to this PR, but why is this not `ZZRingElem` by default?

File: src/Groups/group_characters.jl [hunk 2]
```diff
@@ -1378,6 +1425,116 @@ function class_positions_of_kernel(list::Vector{T}) where T
...
+    class_positions_of_center(chi::GAPGroupClassFunction)
+
+Return the array of those integers `i` such that `chi[i]` is `chi[1]` times
+a root of unity.
+
+# Examples
+```jldoctest
+julia> println(class_positions_of_center(character_table("2.A5")[2]))
+[1, 2]
+
```
Comment (thofma) [inline src/Groups/group_characters.jl]:
```suggestion
```

**Comment by thofma** [comment]
Thanks!

File: src/Groups/group_characters.jl (hunk 2 above)
Comment (ThomasBreuer) [reply src/Groups/group_characters.jl]: O.k.

File: src/Groups/group_characters.jl (hunk 1 above)
Comment (ThomasBreuer) [reply src/Groups/group_characters.jl:1214]: `GAPGroupClassFunction` objects are more general than (virtual) characters, one can think of scalar multiples of differences of characters. One example where rational multiples occur in practice are the coefficient vectors of idempotents in the centre of the group algebra.
Of course one can argue that calling `degree` for such a class function does not make sense. Or one can argue that one could consider arbitrary cyclotomics for scalar multiplication of class functions.

**Review by fingolfin — APPROVED** [review]

**Pull request merged**